subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -98,53 +98,46 @@ static struct bt_mesh_scene_srv *srv_find(uint16_t elem_idx)
 	return NULL;
 }
 
-static uint16_t current_scene(const struct bt_mesh_scene_srv *srv, int64_t now)
+static uint16_t current_scene(const struct bt_mesh_scene_srv *srv)
 {
-	if (bt_mesh_model_transition_time(&srv->transition) &&
-	    srv->prev != srv->next && now < srv->transition_end) {
-		if (now < srv->transition_end - srv->transition.time) {
+	if (k_work_delayable_is_pending(&srv->work)) {
+		/* When we're in a transition, the current scene should be NONE (see the Bluetooth
+		 * mesh model specification, section 5.1.3.2.1). Check that we're not just in a
+		 * delay phase:
+		 */
+		if (!srv->transition.delay ||
```

Comment:
It seems like return values should be swapped between lines 111 and 113.

Reply:
Good catch